-code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-47641/44178

ERROR: Build errors found during clang-tidy run.
```
src/EventFilter/Utilities/src/FedRawDataInputSource.cc:730:65: error: expected ')' [clang-diagnostic-error]
  730 |             if (fileLSOpen && (!fms_ || !fms_->streamIsIdle(i)) {
      |                                                                 ^
src/EventFilter/Utilities/src/FedRawDataInputSource.cc:730:16: note: to match this '('
--
src/EventFilter/Utilities/src/FedRawDataInputSource.cc:734:11: error: expected statement [clang-diagnostic-error]
  734 |           }
      |           ^
Suppressed 2082 warnings (2078 in non-user code, 4 NOLINT).
--
gmake: *** [config/SCRAM/GMake/Makefile.coderules:129: code-checks] Error 2
gmake: *** [There are compilation/build errors. Please see the detail log above.] Error 2
```

--- a/EventFilter/Utilities/interface/FastMonitoringService.h
+++ b/EventFilter/Utilities/interface/FastMonitoringService.h
@@ -233,6 +233,7 @@ namespace evf {
     void setTMicrostate(FastMonState::Microstate m);
 
     static unsigned int getTID() { return tbb::this_task_arena::current_thread_index(); }
+    bool streamIsIdle(unsigned int i) const;
 
   private:
     void doSnapshot(const unsigned int ls, const bool isGlobalEOL);

--- a/EventFilter/Utilities/src/DAQSource.cc
+++ b/EventFilter/Utilities/src/DAQSource.cc
@@ -649,7 +649,7 @@ void DAQSource::fileDeleter() {
         for (unsigned int i = 0; i < streamFileTracker_.size(); i++) {
           if (it->first == streamFileTracker_.at(i)) {
             //only skip if LS is open
-            if (fileLSOpen) {
+            if (fileLSOpen && (!fms_ || !fms_->streamIsIdle(i))) {
               fileIsBeingProcessed = true;
               break;
             }

--- a/EventFilter/Utilities/src/FastMonitoringService.cc
+++ b/EventFilter/Utilities/src/FastMonitoringService.cc
@@ -1052,4 +1052,9 @@ namespace evf {
       fmt_->jsonMonitor_->snap(ls);
   }
 
+  bool FastMonitoringService::streamIsIdle(unsigned int i) const {
+    auto ms = microstate_.at(i);
+    return ms == getmIdle();
+  }
+
 }  //end namespace evf

--- a/EventFilter/Utilities/src/FedRawDataInputSource.cc
+++ b/EventFilter/Utilities/src/FedRawDataInputSource.cc
@@ -727,7 +727,7 @@ void FedRawDataInputSource::fileDeleter() {
         for (unsigned int i = 0; i < streamFileTracker_.size(); i++) {
           if (it->first == streamFileTracker_.at(i)) {
             //only skip if LS is open
-            if (fileLSOpen) {
+            if (fileLSOpen && (!fms_ || !fms_->streamIsIdle(i))) {
               fileIsBeingProcessed = true;
               break;
             }

